common/listener.go

```diff
@@ -55,6 +56,7 @@ func (l *Listener) StartTLS(errChan chan<- error, tlsConfig *tls.Config, address
 	}
 
 	l.tlsListener = list
+	l.tlsLocalAddress = address
```

Comment:
Is this only used for the log at L105? It looks redundant and I don't think we should add a struct field for it since we are logging it here at L60
> log.Printf("[Listener] Started listening on tls endpoint %s.", address)

Reply:
yes, it is used for that. i'm thinking that if there is an absence of the log, it'd cause confusion in the logs if it were missing

Reply:
without the struct field, perhaps this is better: `log.Printf("[Listener] Stopped listening on tls endpoint %s", l.tlsListener.Addr())` instead. i could also convert L113 to `l.listener.Addr()`.

you get log lines kinda like this when running locally:
```
2024/06/07 11:48:23 [11816] [Listener] Started listening on tls endpoint :10091.
2024/06/07 11:48:23 [11816] [Listener] Started listening on :10090.
2024/06/07 11:48:23 [11816] [Listener] Stopped listening on tls endpoint [::]:10091
2024/06/07 11:48:23 [11816] [Listener] Stopped listening on [::]:10090
```

any thoughts?

Reply:
IMO pulling the data out of the listener like this is better than adding a field for it 👍

Reply:
refactored this as well, thanks for suggestion